### PR TITLE
Deprecate CallbackGuard

### PR DESCRIPTION
**Pull request by EPashkin**
Part of https://github.com/gtk-rs/glib/pull/250

cc @GuillaumeGomez

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
👍

**Comment by sdroege** [comment]
:+1:

**Comment by EPashkin** [comment]
Oh, it since 1.23. I fix deprecation text

**Comment by EPashkin** [comment]
Updated

File: src/source.rs [hunk 1]
```diff
@@ -54,20 +54,24 @@ impl ToGlib for Continue {
 
 /// Unwinding propagation guard. Aborts the process if destroyed while
 /// panicking.
+#[deprecated(note="Rustc has this functionality buildin since 1.23.0")]
```
Comment (sdroege) [inline src/source.rs]: Since 1.24

**Comment by EPashkin** [comment]
https://github.com/rust-lang/rust/commit/51b47dc4a1af3260738aa7c5d4e31e8d77c0c0b6 contains in branches started from https://github.com/rust-lang/rust/commit/d0f1f4281632386ab0e3f35eed87fac0e5b2d771 and https://github.com/rust-lang/rust/commit/2dad872a2dae12b882cd73536a66c9289a2b65ae but not in https://github.com/rust-lang/rust/commit/5041b3bb3d953a14f32b15d1e41341c629acae12

Strange that for me https://github.com/rust-lang/rust/releases/tag/1.23.0 not in current repo

**Comment by EPashkin** [comment]
@sdroege Maybe you right, I use "Aaron Power" commits

**Comment by sdroege** [comment]
Also it's in the 1.24 release notes

**Comment by EPashkin** [comment]
You right, I checked by tag archive too. I will change back

**Comment by EPashkin** [comment]
Updated

**Comment by sdroege** [comment]
Great :)

**Comment by sdroege** [comment]
From my side yes

File: src/source.rs [hunk 2]
```diff
@@ -54,20 +54,24 @@ impl ToGlib for Continue {
 
 /// Unwinding propagation guard. Aborts the process if destroyed while
 /// panicking.
+#[deprecated(note="Rustc has this functionality buildin since 1.24.0")]
```
Comment (sdroege) [inline src/source.rs]: "builtin" or "built-in"

Comment (EPashkin) [reply src/source.rs]: Thanks, fixed

**Comment by EPashkin** [comment]
For me PR also ready for merge

**Comment by GuillaumeGomez** [comment]
Thanks then!

**Pull request merged**